staging/src/k8s.io/apiserver/pkg/audit/policy/checker_test.go

```diff
@@ -152,64 +152,198 @@ func TestChecker(t *testing.T) {
 		},
 	}
 
-	test := func(req string, expLevel audit.Level, expOmitStages []audit.Stage, ruleNames ...string) {
-		policy := audit.Policy{}
+	test = func(t *testing.T, req string, expLevel audit.Level, policyStages, expOmitStages []audit.Stage, ruleNames ...string) {
+		policy := audit.Policy{OmitStages: policyStages}
 		for _, rule := range ruleNames {
 			require.Contains(t, rules, rule)
 			policy.Rules = append(policy.Rules, rules[rule])
 		}
 		require.Contains(t, attrs, req)
 		actualLevel, actualOmitStages := NewChecker(&policy).LevelAndStages(attrs[req])
 		assert.Equal(t, expLevel, actualLevel, "request:%s rules:%s", req, strings.Join(ruleNames, ","))
-		assert.Equal(t, expOmitStages, actualOmitStages, "request:%s rules:%s", req, strings.Join(ruleNames, ","))
+		assert.True(t, stageEqual(expOmitStages, actualOmitStages), "request:%s rules:%s, expected stages: %v, actual stages: %v",
+			req, strings.Join(ruleNames, ","), expOmitStages, actualOmitStages)
+	}
+)
+
+func TestChecker(t *testing.T) {
+	test(t, "namespaced", audit.LevelMetadata, nil, nil, "default")
```

Comment:
What I had in mind is moving all the copy-pasta to one method, to write this long list of methods once, not twice

Reply:
Yeah, we can do this.
But we have to do some work on the 4th, 5th parameter of test() function.

Like this:
```
policyStages = nil // or policyStages = []audit.Stage{"foo", "bar"}
test(t, "namespaced", audit.LevelNone, unionStages(nil, policyStages), unionStages(nil, policyStages), "create")
```

I think it's not so good to use function `unionStages` to test itself.

Reply:
Why can't you pass parameters `foo` to replace `nil` in one case and `policyStages` in another case?

Reply:
`policyStages` is not used every where.
For example:
https://github.com/kubernetes/kubernetes/pull/54634/files/46e4d37cdb65aea9308f4656941b4a8a39302361#diff-f43e8742cbd284648301420a7833503bR267

I need to combine them first with function unionStages().
It's not so good to use function unionStages to test itself.

Reply:
I know, but all occurrences of either `nil` or `policyStages` can be replaced with a function parameter

Reply:
Basically you have the following:

```go
func foo() {
  f(1, nil, nil)
  f(1, nil, nil)
  f(1, 2, nil)
}

func bar() {
  f(1, policyStages, policyStages)
  f(1, policyStages, policyStages)
  f(1, 2, policyStages)
}
```

What you can do instead:

```go
func foo() {
  baz(nil)
}

func bar() {
  baz(policyStages)
}

func baz(x X) {
  f(1, x, x)
  f(1, x, x)
  f(1, 2, x)
}
```

Reply:
Done.